### ISSUE TITLE
Hovering over field causes value to disappear

### ISSUE THREAD

**Issue by sabr49** [RobinHerbots/Inputmask#334]
This may be an edge case but is common in our application... for business reason we have fields that are disabled on load and have their property values set using $.val... If later the field is enabled and then you hover over the field the entire input is cleared...

Here is the fiddle:
http://jsfiddle.net/fctWm/1/

Hover over the input..

I'm working with an old version (really old...  1.0.24 :) ) but essentially the fix I've found lies here (I know this has since been updated - however the fiddle is using the latest commit)... 

```
function installEventRuler(npt) {
                    var events = $._data(npt).events;

                    $.each(events, function (eventType, eventHandlers) {
                        if(eventType == 'setvalue') return; /* my line here */
                        $(npt).bind(eventType + ".inputmask", function (event) {
                            if (this.readOnly || this.disabled) {
                                event.stopPropagation();
                                event.stopImmediatePropagation();
                                event.preventDefault();
                                return false;
                            }
                        });
                        //!! the bound handlers are executed in the order they where bound
                        //reorder the events
                        var ourHandler = eventHandlers[eventHandlers.length - 1];
                        for (var i = eventHandlers.length - 1; i > 0; i--) {
                            eventHandlers[i] = eventHandlers[i - 1];
                        }
                        eventHandlers[0] = ourHandler;
                    });
                }

```

It seems we are preventing all of our handlers, including the setvalue handler from firing which doesn't allow the undobuffer variable to be set - not sure that it makes sense for setvalue..

**Comment by RobinHerbots** [comment]
Hi Sabr,

On http://jsfiddle.net/fctWm/2/ i modified the inputmask to allow setting
the value on a disabled input.

It will be available in version 2.3.41.

If you add an extra check in the eventruler to not block the setvalue
event, it will also work on the 1.0.24.  But I don't have the time to
maintain the 1.x branch.

Best regards,

Robin

2013/9/12 sabr49 notifications@github.com

**Comment by RobinHerbots** [comment]
@sabr49, 

See http://jsfiddle.net/fctWm/5/ with the latest version of inputmask
